### PR TITLE
test: set up `defaults` test helper to run a test per use case

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #N/A

## Summary
Refactor `defaults` test helper to help mitigate timeouts.
The updated doc reflects how to use the helpers in E2E tests.

### CODE REVIEW

**Comment by Elijbet** [comment]
I can't seem to prevent the linter from removing the empty lines in comments (it overrides the existing ones). I'm going to look into which rule from the JSDoc plugin might be causing the issue, ignoring it for now since it's unrelated to my changes. Although the formatting of the examples in the comments doesn't look great.

**Comment by driskull** [comment]
Can you open a separate PR with the suggested linting changes? We can tag Franco/Ben for their opinion.

**Review by driskull — APPROVED** [review]
👍 Nice!

**Review by benelan — COMMENTED** [review]
Created a PR for the blank line rule, nice find #7018 

Not sure if we need the indent rule, see my other comment

File: src/tests/commonTests.ts [hunk 1]
```diff
@@ -155,26 +152,42 @@ function propToAttr(name: string): string {
 /**
  * Helper for asserting that a property's value is its default
  *
+ * Note that this helper should be used within a describe block.
+ *
```
[the remark this answers is not in the document]
Comment (jcfranco) [reply src/tests/commonTests.ts:156]: Good catch. We should do that for all JSDoc code snippets. I know I've omitted them sometimes. 😬

**Review by jcfranco — APPROVED** [review]
💥

**Review by jcfranco — APPROVED** [review]

**Pull request merged**